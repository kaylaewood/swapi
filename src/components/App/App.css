.clearfix:after {
  clear: both;
  content: ' ';
  display: table;
}

*,
*:after,
*:before {
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  margin: 0;
  padding: 0;
}

.App {
  background-attachment: fixed;
  background-color: black;
  background-image: url('../../images/background.jpg');
  background-size: cover;
  height: 80vw;
}
