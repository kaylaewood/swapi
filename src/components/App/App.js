import React, { Component } from 'react';
import './App.css';
import Header from '../Header/Header';
import Form from '../Form/Form';
import UserProfile from '../UserProfile/UserProfile';
import MovieContainer from '../MovieContainer/MovieContainer';
import CharacterContainer from '../CharacterContainer/CharacterContainer';
import { Route, Redirect } from 'react-router-dom';

class App extends Component {
  constructor() {
    super();
    this.state = {
      loggedIn: false,
      name: '',
      favoriteQuote: '',
      ranking: '',
      movies: [],
      characters: []
    }
  }

  componentDidMount() {
    fetch('https://swapi.co/api/films')
    .then(response => response.json())
    .then(movies => {
      this.setState({
        movies: movies.results
      })
    })
    .catch(error => console.log(error))
  }

  updateStateFromForm = (name, favoriteQuote, ranking) => {
    this.setState({
      loggedIn: true,
      name: name,
      favoriteQuote: favoriteQuote,
      ranking: ranking
    })
  }

  updateStateWithCharacters = (character) => {
    if (this.state.characters.length < 10) {
      let characterList = [...this.state.characters, {
        name: character.name,
        species: character.species,
        homeworld: character.homeworld,
        homeworldPop: 'TBD',
        relatedFilms: character.films
      }];
      this.setState({
        characters: characterList
      })
    }
  }

  fetchCharacters = (movie) => {
    movie.characters.forEach(character => fetch(character)
      .then(response => response.json())
      .then(data => {
        console.log(data);
        this.updateStateWithCharacters(data)
      })
      .catch(error => console.log(error))
    )
  }


  render() {
    return (
      <main className="App">
        <Route exact path="/">
          {this.state.loggedIn ? <Redirect to="/movies" /> : (
            <>
              <Header heading='HELLO, YOUNG JEDI'/>
              <Form
                updateStateFromForm={this.updateStateFromForm}
              />
            </>
          )}
        </Route>
        <Route exact path='/movies' render={() => {
          return (
            <>
              <UserProfile
                name={this.state.name}
                favoriteQuote={this.state.favoriteQuote}
                ranking={this.state.ranking}
              />
              <Header heading='THE MOVIES'/>
              <MovieContainer
                movies={this.state.movies}
                fetchCharacters={this.fetchCharacters}
              />
            </>
          )
        }} />
        <Route path='/movies/:id' render={({ match }) => {
          console.log(match.params.id);
          return (
            <>
              <UserProfile
                name={this.state.name}
                favoriteQuote={this.state.favoriteQuote}
                ranking={this.state.ranking}
              />
              <Header heading='THE CHARACTERS'/>
              <CharacterContainer characters={this.state.characters}/>
            </>
          )
        }} />
      </main>
    );
  }
};

export default App;
