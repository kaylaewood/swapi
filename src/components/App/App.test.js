import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() {
    wrapper = shallow(<App />)

  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state with form information', () => {

  })

  it('should update state with character URLs', () => {

  })

  it('should update state when user logs out', () => {
    
  })
})
