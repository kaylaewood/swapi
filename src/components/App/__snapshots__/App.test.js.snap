// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the favorited class 1`] = `
<main
  className="App"
>
  <Header />
  <Form />
</main>
`;
